### ISSUE TITLE
Next. js version 12.3.4, invalid server report, no errors collected；"@sentry/nextjs": "8.13.0"

### ISSUE THREAD

**Issue by parlay96** [getsentry/self-hosted#3183]
### Environment

self-hosted (https://develop.sentry.dev/self-hosted/)

### Steps to Reproduce

1： Firstly, the register method on the server was not executed at all，
![image](https://github.com/getsentry/sentry/assets/32978177/0577d7df-0fba-4469-84b6-0e243a7a77d1)
2： The Sentry platform also did not collect any errors regarding the server side
![image](https://github.com/getsentry/sentry/assets/32978177/a84cb59c-6988-4312-9217-379caad52237)
3：Only client errors are effective for the client
![image](https://github.com/getsentry/sentry/assets/32978177/afb143bd-97f3-4c34-9cf3-91e2dfd117e9)

### Expected Result

How to solve it?

### Actual Result

1： Firstly, the register method on the server was not executed at all，
![image](https://github.com/getsentry/sentry/assets/32978177/0577d7df-0fba-4469-84b6-0e243a7a77d1)
2： The Sentry platform also did not collect any errors regarding the server side
![image](https://github.com/getsentry/sentry/assets/32978177/a84cb59c-6988-4312-9217-379caad52237)
3：Only client errors are effective for the client
![image](https://github.com/getsentry/sentry/assets/32978177/afb143bd-97f3-4c34-9cf3-91e2dfd117e9)

### Product Area

Unknown

### Version

 "@sentry/nextjs": "8.13.0"

**Comment by aldy505** [comment]
Can you confirm that this is only happening with Nextjs and this is not happening on any other language/platforms?

**Comment by parlay96** [comment]
@aldy505 I'm certainly not sure, but there's this situation next，However, what I have learned is that the next version on my end is 12.xx, which is a version issue

**Comment by aldy505** [comment]
I don't think this has anything to do with self-hosted. Since Sentry should receive all the data, I guess this is much likely a problem on the SDK itself.

@hubertdeng123 @azaslavsky can we transfer this to sentry-javascript?
